### DISCUSSION TITLE
Document Scanning

### DISCUSSION THREAD

**Discussion by SteveUKuser** [elementary/triage#166] [Q&A]
I have just installed Elementary OS 7 on an old Samsung laptop.
Its working well but for continued use by me I need to be able to import a scan from my HP 4130e.
The wireless printer is detected and works. but no device is detected that is a scanner.
I have now connected the 4130e directly via USB.
I have run scanimage -L from the terminal prompt.
The result is device escl:https://192.168.1.174:443 is a HP Deskjet 4100 series(usb) platen adf scanner.
HP-setup reports that it requires GUI support and that PyQt4 and PyQt5 are not installed.
How do I install the relevant drivers so that Elementary can be used with this scanner.
Thanks for all your help with this issue.

**Comment by jeremypw** [comment]
Getting printers and scanners to work properly with Linux can be a bit hit and miss as it sometimes depends on the availability of suitable drivers from the manufacturer.  I use a Brother Laser printer/scanner and they helpfully supply a driver installation script and instructions on their website.  It works well with the "Document Scanner" program from the AppCenter when plugged into USB.  I see the HP website does have fairly extensive information on Linux drivers.  Hopefully someone with the same printer who has succeeded in setting it up for scanning can help further.

**Comment by jonnjonzzn** [comment]
As @jeremypw mentioned printing/MFP in general can be an adventure. I would suggest googling: "ubuntu 22.04 hplip" which is the name of the set of drivers from HP for their printers and multi-function devices on Linux.  They are available via the standard available elementaryos repositories which is likely the easiest path although HP will likely have newer version available on its [site](https://developers.hp.com/hp-linux-imaging-and-printing). 

As a first attempt you might try this:
Start Terminal:
$sudo apt install hplip-gui

Then go look for "HPLIP ToolBox" in Applications.  Good luck!
